src/op_handlers/opcode_decommit.rs

```diff
@@ -20,7 +20,9 @@ pub fn opcode_decommit(
 
     vm.decrease_gas(extra_cost)?;
 
-    let code = storage
+    let code = world
+        .contracts_storage
+        .borrow()
```

Comment:
Same comment as above, this borrow will panic if there is an existing mutable reference to code, or a new one is created, let's wrap this in the 'decommit' method I proposed above, or something else, let's just make sure we drop the ref.